clients/fides-js/src/components/tcf/DoubleToggleTable.tsx

```diff
@@ -0,0 +1,166 @@
+/** This isn't a table _yet_, but it is aspirationally named.
+ * We should probably convert it to a table at some point fides#4190 */
+
+import { VNode, h } from "preact";
+import { useMemo } from "react";
+import DataUseToggle from "../DataUseToggle";
+import Toggle from "../Toggle";
+import { EnabledIds, LegalBasisEnum } from "../../lib/tcf/types";
+import type { UpdateEnabledIds } from "./TcfOverlay";
+
+interface Item {
+  id: string | number;
+  name?: string;
+  isConsent: boolean;
+  isLegint: boolean;
+}
+
+interface Props<T extends Item> {
+  items: T[];
+  title: string;
+  enabledConsentIds: string[];
+  enabledLegintIds: string[];
+  renderToggleChild: (item: T) => VNode;
+  onToggle: (payload: UpdateEnabledIds) => void;
+  renderBadgeLabel?: (item: T) => string | undefined;
+  consentModelType: keyof EnabledIds;
+  legintModelType: keyof EnabledIds;
+}
+
+const DoubleToggleTable = <T extends Item>({
```

Comment:
typescript templating!! super handy for this case, where I wanted the table to handle both vendors and purposes

Reply:
otherwise, this file is mostly the same as what used to be in `TcfVendors`, just generic'd a bit